src/main/kotlin/domain/step3/domain/game/RacingCarGameRecord.kt

```diff
@@ -0,0 +1,19 @@
+package domain.step3.domain.game
+
+import domain.step3.domain.racingcar.RacingCars
+
+@JvmInline
+value class RacingCarGameRecord(private val _racingCarGameRecord: List<RacingCars>) {
+    val racingCarGameRecord: List<RacingCars>
+        get() = _racingCarGameRecord.toList()
+
+    fun add(racingCars: RacingCars): RacingCarGameRecord {
+        val mutableRecord = _racingCarGameRecord.toMutableList()
+        mutableRecord.add(racingCars)
+        return RacingCarGameRecord(mutableRecord)
+    }
```

Comment:
kotlin은 연산자 오버로딩이 가능합니다.
그래서 Collection에 해당하는 연산자 오버로딩들이 kotlin api에 많이 구현되어있습니다.

```suggestion
    fun add(racingCars: RacingCars): RacingCarGameRecord {
        return RacingCarGameRecord(_racingCarGameRecord + racingCars)
    }
```
이런식으로 좀 더 코틀린스럽게 작성할 수 있습니다.

Reply:
현재 코틀린 인 액션을 정독중으로써,연산자 오버로딩이 있다는 것만 알지 이렇게 편리하게 사용될 줄은 몰랐네요
정말 감사합니다 :) 얼른 공부해서 자유롭게 적용해볼 수 있도록 해보겠습니다.